### ISSUE TITLE
Intel_adsp_cavs18: test cases can not get complete log

### ISSUE THREAD

**Issue by LixinGuoX** [zephyrproject-rtos/zephyr#41993]
**Describe the bug**
For test cases, binary image can be downloaded to ADSP, and there is some output can be caught by adsplog.py, but the log is not complete.

**To Reproduce**
use twister command:
`twister -p intel_adsp_cavs18 --device-testing -T tests/ --west-flash="/home/ztest/zephyrproject/UP_WHL/tools/flash18.sh" --device-serial-pty="/home/ztest/zephyrproject/UP_WHL/tools/adsplog.sh,--no-history" -j 1`
here is the flash18.sh and adsplog.sh:
```
flash18.sh
#!/bin/sh

BUILD=$1
FIRMWARE=${BUILD}/zephyr/zephyr.ri
FLASHER1=/home/ztest/zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs15/tools/flash.sh
FLASHER=/home/ztest/zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs15/tools/cavs-fw-v15.py
KEY=/home/ztest/zephyrproject/modules/audio/sof/keys/otc_private_key.pem
CONFIG=/home/ztest/zephyrproject/modules/audio/sof/rimage/config
DIR=/home/ztest/zephyrproject/modules/audio/sof/rimage/build/rimage

echo "Signing with key " $key ${BUILD}
west sign -d ${BUILD} -t rimage -p ${DIR} -D ${CONFIG} -- -k ${KEY} --no-manifest

echo ${FLASHER} ${FIRMWARE}
timeout 60 ssh root@localhost ${FLASHER} ${FIRMWARE}

ssh root@localhost pkill -f -9 cavs-fw-v15.py || true

```
```
adsplog.sh
#!/bin/sh

ADSPLOG=/home/ztest/zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs15/tools/adsplog.py

ssh root@localhost pkill -f -9 adsplog.py || true

timeout 300 ssh root@localhost python3 -u ${ADSPLOG} $1
```

**Impact**
The log indicates a problem, it seems that it is too fast for adsplog.py to get the log.

**Logs and console output**
example log as below:
```
DEBUG   - Running cmake on /home/ztest/zephyrproject/zephyr/tests/kernel/mutex/mutex_api for intel_adsp_cavs18
DEBUG   - Calling cmake with arguments: ['-B/home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex', '-S/homgs', '-DEXTRA_LDFLAGS=-Wl,--fatal-warnings', '-DEXTRA_GEN_DEFINES_ARGS=--edtlib-Werror', '-GUnix Makefiles', '-DBOARD=intel_adsp_cavs18']
DEBUG   - Finished building /home/ztest/zephyrproject/zephyr/tests/kernel/mutex/mutex_api for intel_adsp_cavs18
DEBUG   - build test: intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex
DEBUG   - Building /home/ztest/zephyrproject/zephyr/tests/kernel/mutex/mutex_api for intel_adsp_cavs18
DEBUG   - run test: intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex
DEBUG   - Using serial device /dev/pts/49 @ 115200 baud
DEBUG   - Flash command: ['west', 'flash', '--skip-rebuild', '-d', '/home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/kernel/mutex/mutex_ap
DEBUG   - DEVICE: Killed
DEBUG   - DEVICE: access resource from main thread
DEBUG   - DEVICE: PASS - test_mutex_reent_lock_forever in 1.1 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_mutex_reent_lock_no_wait
DEBUG   - DEVICE: access resource from main thread
DEBUG   - DEVICE: bypass locked resource from spawn thread
DEBUG   - -- west flash: using runner misc-flasher
Signing with key  /home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex
Signing for SOC target cnl using cnl.toml
Signing with tool /home/ztest/zephyrproject/modules/audio/sof/rimage/build/rimage
/home/ztest/zephyrproject/modules/audio/sof/rimage/build/rimage -k /home/ztest/zephyrproject/modules/audio/sof/keys/otc_private_key.pem -o /home/ztest/zephyrpect/modules/audio/sof/rimage/config/cnl.toml -i 3 /home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex/zepn.mod

Module Reading /home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex/zephyr/boot.mod
info: ignore .bss section for bootloader module
  Found 6 sections, listing valid sections......
        No      LMA             VMA             End             Size    Type    Name
        0       0x00020000      0x00000000      0x00000000      0x0
        1       0xb0038000      0xb0038000      0xb0038364      0x364   TEXT    .imr

 module: input size 868 (0x364) bytes 1 sections
 module: text 868 (0x364) bytes
    data 0 (0x0) bytes
    bss  0 (0x0) bytes


Module Reading /home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex/zephyr/main.mod
  Found 42 sections, listing valid sections......
        No      LMA             VMA             End             Size    Type    Name
        1       0xbe010000      0xbe010000      0xbe01016a      0x16a   TEXT    .WindowVectors.text
        2       0xbe010180      0xbe010180      0xbe0101a9      0x29    TEXT    .Level2InterruptVector.text
        3       0xbe0101c0      0xbe0101c0      0xbe0101e9      0x29    TEXT    .Level3InterruptVector.text
        4       0xbe010200      0xbe010200      0xbe010229      0x29    TEXT    .Level4InterruptVector.text
        5       0xbe010240      0xbe010240      0xbe010269      0x29    TEXT    .Level5InterruptVector.text
        6       0xbe010280      0xbe010280      0xbe0102a9      0x29    TEXT    .DebugExceptionVector.text
        7       0xbe0102c0      0xbe0102c0      0xbe0102e9      0x29    TEXT    .NMIExceptionVector.text
        8       0xbe010300      0xbe010300      0xbe010303      0x3     TEXT    .KernelExceptionVector.text
        9       0xbe010340      0xbe010340      0xbe010352      0x12    TEXT    .UserExceptionVector.text
        10      0xbe0103c0      0xbe0103c0      0xbe0103c6      0x6     TEXT    .DoubleExceptionVector.text
        11      0xbe010400      0xbe010400      0xbe017daa      0x79aa  TEXT    .text
        12      0xbe018000      0xbe018000      0xbe019c54      0x1c54  DATA    .rodata
        13      0xbe019c54      0xbe019c54      0xbe019c94      0x40    DATA    initlevel
        14      0xbe019c94      0xbe019c94      0xbe019cf4      0x60    DATA    devices
        15      0xbe019cf4      0xbe019cf4      0xbe019d0c      0x18    DATA    log_const_sections
        16      0xbe019d0c      0xbe019d0c      0xbe019d1c      0x10    DATA    log_backends_sections
        17      0xbe019d1c      0xbe019d1c      0xbe019d34      0x18    DATA    device_handles
        18      0xbe019d34      0xbe019d34      0xbe01cb40      0x2e0c  DATA    .noinit
        19      0xbe01cb40      0xbe01cb40      0xbe01cc08      0xc8    DATA    .data
        20      0xbe01cc08      0xbe01cc08      0xbe01d0b0      0x4a8   DATA    sw_isr_table
        21      0xbe01d0b0      0xbe01d0b0      0xbe01d0c0      0x10    DATA    device_states
        22      0xbe01d0c0      0xbe01d0c0      0xbe01d0d4      0x14    DATA    k_mutex_area
        23      0xbe01d0d4      0xbe01d0d4      0xbe01d0e4      0x10    DATA    k_sem_area
        24      0xbe01e000      0xbe01e000      0xbe01e928      0x928   BSS     .bss

 module: input size 52489 (0xcd09) bytes 24 sections
 module: text 31781 (0x7c25) bytes
    data 20708 (0x50e4) bytes
    bss  2344 (0x928) bytes

Module Write: /home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex/zephyr/boot.mod
 Manifest module metadata section at index 2
 Entry point 0xb0038000

        Totals  Start           End             Size
        TEXT    0xb0038000      0xb0038364      0x364
        DATA    0xffffffff      0x00000000      0x1
        BSS     0x00000000      0x00000000      0x0

        No      Address         Size            File    Type
        1       0xb0038000      0x364           0x8000  TEXT

 Total pages text 1 data 1 bss 0 module file limit: 0x9000

Module Write: /home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex/zephyr/main.mod
 Manifest module metadata section at index 25
 Entry point 0xbe010400

        Totals  Start           End             Size
        TEXT    0xbe010000      0xbe017daa      0x7daa
        DATA    0xbe018000      0xbe01d0e4      0x50e4
        BSS     0xbe01e000      0xbe01e928      0x928

        No      Address         Size            File    Type
        1       0xbe010000      0x16a           0x9000  TEXT
        2       0xbe010180      0x29            0x9180  TEXT
        3       0xbe0101c0      0x29            0x91c0  TEXT
        4       0xbe010200      0x29            0x9200  TEXT
        5       0xbe010240      0x29            0x9240  TEXT
        6       0xbe010280      0x29            0x9280  TEXT
        7       0xbe0102c0      0x29            0x92c0  TEXT
        8       0xbe010300      0x3             0x9300  TEXT
        9       0xbe010340      0x12            0x9340  TEXT
        10      0xbe0103c0      0x6             0x93c0  TEXT
        11      0xbe010400      0x79aa          0x9400  TEXT
        12      0xbe018000      0x1c54          0x11000 DATA
        13      0xbe019c54      0x40            0x12c54 DATA
        14      0xbe019c94      0x60            0x12c94 DATA
        15      0xbe019cf4      0x18            0x12cf4 DATA
        16      0xbe019d0c      0x10            0x12d0c DATA
        17      0xbe019d1c      0x18            0x12d1c DATA
        18      0xbe019d34      0x2e0c          0x12d34 DATA
        19      0xbe01cb40      0xc8            0x15b40 DATA
        20      0xbe01cc08      0x4a8           0x15c08 DATA
        21      0xbe01d0b0      0x10            0x160b0 DATA
        22      0xbe01d0c0      0x14            0x160c0 DATA
        23      0xbe01d0d4      0x10            0x160d4 DATA

 Total pages text 8 data 6 bss 1 module file limit: 0x17000

Firmware completing manifest v1.8
 meta: completing ADSP manifest
 meta: limit is 0x18b80
 auth: completing authentication manifest
 cse: ri_css_v1_8_hdr_create completing CSS manifest
 css: set build date to 2022:00:20
 cse: completing CSE V1.8 manifest
Firmware file size 0x1d000 page count 23
 rimage_read_key: read key '/home/ztest/zephyrproject/modules/audio/sof/keys/otc_private_key.pem'
 pkcs: RSA private key is valid.
 pkcs: digest for manifest is 8e73b552b4f5d996e262fee4e9f728012416cb8e137e51f622d5ed08291cef53
Firmware manifest and signing completed !
/home/ztest/zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs15/tools/cavs-fw-v15.py /home/ztest/zephyrproject/zephyr/twister-out/intel_adsp_cavs18/tests/ker

DEBUG   - DEVICE: access resource from main thread
DEBUG   - DEVICE: bypass locked resource from spawn thread
DEBUG   - DEVICE: PASS - test_mutex_reent_lock_no_wait in 1.1 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_mutex_reent_lock_timeout_fail
DEBUG   - DEVICE: access resource from main thread
DEBUG   - DEVICE: bypass locked resource from spawn thread
DEBUG   - DEVICE: access resource from main thread
DEBUG   - DEVICE: bypass locked resource from spawn thread
DEBUG   - DEVICE: PASS - test_mutex_reent_lock_timeout_fail in 2.1 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_mutex_reent_lock_timeout_pass
DEBUG   - DEVICE: access resource from main thread
DEBUG   - DEVICE: access resource from spawn thread
DEBUG   - DEVICE: access resource from main thread
DEBUG   - DEVICE: bypass locked resource from spawn thread
DEBUG   - DEVICE: PASS - test_mutex_reent_lock_timeout_pass in 2.1 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_mutex_recursive
DEBUG   - DEVICE: PASS - test_mutex_recursive in 0.2 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_mutex_priority_inheritance
DEBUG   - DEVICE: PASS - test_mutex_priority_inheritance in 6.1 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: Test suite mutex_api succeeded
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: PROJECT EXECUTION SUCCESSFUL
DEBUG   - Process /home/ztest/zephyrproject/UP_WHL/tools/adsplog.sh,--no-history terminated outs: None errs None
DEBUG   - run status: intel_adsp_cavs18/tests/kernel/mutex/mutex_api/kernel.mutex passed
INFO    - 40/87 intel_adsp_cavs18         tests/kernel/mutex/mutex_api/kernel.mutex          PASSED (device 13.215s)

```

**Comment by teburd** [comment]
@LixinGuoX can you please reverify with the updated cavstwist.sh script @andyross wrote as I think that would fix this particular issue

**Comment by LixinGuoX** [comment]
Yes, this issue is fixed by cavstwist.sh, so close this issue.

**Comment by enjiamai** [comment]
Verified this is fixed by #42020
